language/move-prover/spec-lang/src/translate.rs

```diff
@@ -1609,6 +1609,73 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
 /// ## Type Translation
 
 impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'module_translator> {
+    /// Translates an hlir type into a target AST type.
+    fn translate_hlir_single_type(&mut self, ty: &SingleType) -> Type {
+        use move_lang::hlir::ast::SingleType_::*;
+        match &ty.value {
+            Ref(is_mut, ty) => {
+                Type::Reference(*is_mut, Box::new(self.translate_hlir_base_type(&*ty)))
+            }
+            Base(ty) => self.translate_hlir_base_type(&*ty),
+        }
+    }
+
+    fn translate_hlir_base_type(&mut self, ty: &BaseType) -> Type {
+        use move_lang::{
+            hlir::ast::{BaseType_::*, TypeName_::*},
+            naming::ast::BuiltinTypeName_::*,
+        };
+        match &ty.value {
+            Param(TParam { debug, .. }) => {
```

Comment:
Why called debug?

Reply:
This is a question for @tnowacki .  I asked him and he agreed that a better name should be used.

Reply:
This is a question for @tnowacki .  I asked him and he agreed that a better name should be used.